### ISSUE TITLE
"No Response" on Air Conditioners after a few days / weeks of working fine

### ISSUE THREAD

**Issue by dessmith** [lprhodes/homebridge-broadlink-rm#493]
I have 2 air con units setup (BedAir & LRAir), and working with Homebridge. They've been working for maybe a year with this plugin (which is awesome, so thanks there 😄), but every now and then (most recently after about a week or a few days of being perfectly fine) they would appear as "No Response" in HomeKit, to resolve this I tended to just restart my Raspberry PI, but now I thought it's a good time to just report this and see if I can get some help on it.

They they are shown as no response, I get the following log message for them. "requested temperature from device, waiting" appears to be a warning, however the temperature is shown with the onTemperatue log.

Logs

```
[2019-5-11 18:47:07] [Broadlink RM] BedAir addTemperatureCallbackToQueue (requested temperature from device, waiting)
[2019-5-11 18:47:07] [Broadlink RM] BedAir onTemperature (26.8)
[2019-5-11 18:47:10] [Broadlink RM] LRAir addTemperatureCallbackToQueue (requested temperature from device, waiting)
[2019-5-11 18:47:10] [Broadlink RM] LRAir onTemperature (25.4)
[2019-5-11 18:47:17] [Broadlink RM] BedAir addTemperatureCallbackToQueue (requested temperature from device, waiting)
[2019-5-11 18:47:17] [Broadlink RM] BedAir onTemperature (26.8)
[2019-5-11 18:47:20] [Broadlink RM] LRAir addTemperatureCallbackToQueue (requested temperature from device, waiting)
[2019-5-11 18:47:20] [Broadlink RM] LRAir onTemperature (25.4)
[2019-5-11 18:47:27] [Broadlink RM] BedAir addTemperatureCallbackToQueue (requested temperature from device, waiting)
[2019-5-11 18:47:28] [Broadlink RM] BedAir onTemperature (26.8)
```
My config is

```
{
  "platform": "BroadlinkRM",
  "name": "Broadlink RM",
  "hideScanFrequencyButton": true,
  "hideLearnButton": true,
  "accessories": [
    {
      "name": "Learn Code",
      "type": "learn-code",
      "host": "Static IP2"
    },
    {
      "name": "BedAir",
      "type": "air-conditioner",
      "host": "Static IP1",
      "defaultCoolTemperature": 23,
      "heatTemperature": 30,
      "data": {
        "off": "...",
        "temperature19": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature20": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature21": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature22": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature23": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature24": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature25": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature26": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature27": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature28": {
          "pseudo-mode": "cool",
          "data": "..."
        }
      }
    },
    {
      "name": "LRAir",
      "type": "air-conditioner",
      "host": "Static IP2",
      "defaultCoolTemperature": 24,
      "defaultHeatTemperature": 28,
      "heatTemperature": 30,
      "data": {
        "off": "...",
        "temperature19": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature20": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature21": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature22": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature23": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature24": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature25": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature26": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature27": {
          "pseudo-mode": "cool",
          "data": "..."
        },
        "temperature28": {
          "pseudo-mode": "cool",
          "data": "..."
        }
      }
    }
  ]
}
```

**Comment by stevenleeg** [comment]
I've had a similar issue happen with my IR-based devices. They all stopped working suddenly as I was trying to configure a separate device on Homebridge, and haven't come back since. I've tried restarting homebridge, reconfiguring the Broadlink platform, etc. to no avail.

Any idea as to what might be causing this?

**Comment by stevenleeg** [comment]
Actually, just ended up dedicating some time to fix this and I stumbled upon a solution. I'm still not entirely sure what went wrong, but I was able to fix the problem by removing Homebridge from Homekit, changing the `username` field in my `config.json` to a new value, restarting, and then re-adding Homebridge to Homekit.

For reference, this issue was pretty strange because my IR devices were still working via Alexa (connected through [homebridge-alexa](https://github.com/NorthernMan54/homebridge-alexa)), but showing up as unresponsive on Homekit.
